api/server/creds.go

```diff
@@ -11,61 +12,155 @@ import (
 func (vd *volAPI) credsEnumerate(w http.ResponseWriter, r *http.Request) {
 	method := "credsEnumerate"
 
-	d, err := vd.getVolDriver(r)
+	ctx, err := vd.annotateContext(r)
+
 	if err != nil {
-		notFound(w, r)
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	creds, err := d.CredsEnumerate()
+	// Get gRPC connection
+	conn, err := vd.getConn()
 	if err != nil {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	json.NewEncoder(w).Encode(creds)
+
+	client := api.NewOpenStorageCredentialsClient(conn)
+	resp, err := client.Enumerate(ctx, &api.SdkCredentialEnumerateRequest{})
+
+	if err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	credentials := make(map[string]interface{})
+
+	for _, credentialId := range resp.CredentialIds {
+		credResp, err := client.Inspect(r.Context(), &api.SdkCredentialInspectRequest{
```

Comment:
@stgleb I don't think this is correct. The context in line 15 is what you want. That has the Auth token. Did this work in your tests?

Reply:
Thank you, I've missed that. I'll test it again.

Reply:
Yes, I've missed that case since I did not create any credentials before I test enumerate. My fault.